generic/sosreport.py

```diff
@@ -0,0 +1,145 @@
+            self.is_fail += 1
+            self.log.info("--profile option failed")
+        if 'testname' not in process.system_output("sosreport --batch --case-id=test123 --name=testname  | "
+                                                   "grep tar.xz", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--name option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 --config-file=/etc/sos.conf")
+        file_name = process.system_output("sosreport --batch --case-id=test123 --tmp-dir=/root | "
+                                          "grep root | tail -1", shell=True, ignore_status=True).strip()
+        if not os.path.exists(file_name):
+            self.is_fail += 1
+            self.log.info("--tmp-dir option failed")
+        file_name_bz2 = process.system_output("sosreport --batch --case-id=test123 -z bzip2 | "
+                                              "grep tar.bz2", shell=True, ignore_status=True).strip()
+        if not os.path.exists(file_name_bz2):
+            self.is_fail += 1
+            self.log.info("-z bzip2 option failed")
+        file_name_gz = process.system_output("sosreport --batch --case-id=test123 -z gzip | "
+                                             "grep tar.gz", shell=True, ignore_status=True).strip()
+        if not os.path.exists(file_name_gz):
+            self.is_fail += 1
+            self.log.info("-z gzip option failed")
+        file_name_xz = process.system_output("sosreport --batch --case-id=test123 -z xz | "
+                                             "grep tar.xz", shell=True, ignore_status=True).strip()
+        if not os.path.exists(file_name_xz):
+            self.is_fail += 1
+            self.log.info("-z xz option failed")
+        file_name_xz2 = process.system_output("sosreport --batch --case-id=test123 -z auto | "
+                                              "grep tar.xz", shell=True, ignore_status=True).strip()
+        if not os.path.exists(file_name_xz2):
+            self.is_fail += 1
+            self.log.info("-z auto option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 --chroot=always")
+        self.run_cmd("sosreport --batch --case-id=test123 --chroot=auto")
+        self.run_cmd("sosreport --batch --case-id=test123 --chroot=never")
+        dir_name = process.system_output("sosreport --no-report --batch --build | "
+                                         "grep located | cut -d':' -f2", shell=True, ignore_status=True).strip()
+        sosreport_dir = os.path.join(dir_name, 'sos_reports')
+        if os.listdir(sosreport_dir) != []:
+            self.is_fail += 1
+            self.log.info("--no-report option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -s /home/")
+        process.run("rm -rf /tmp/sosreport*test123*")
+        process.run("rm -rf /root/sosreport*test123*")
```

Comment:
`rm -rf` is dangerous instead can you please use `mkdtemp - for temporary dir` and `mkstemp - for temporary file` and then delete them.